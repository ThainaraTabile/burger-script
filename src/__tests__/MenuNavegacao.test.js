import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { expect } from '@jest/globals';
import '@testing-library/jest-dom/extend-expect';
import MenuNavegacao from '../componentes/MenuNavegacao/MenuNavegacao';

describe('MenuNavegacao', () => {
  test('renderiza o link com o texto e a imagem corretos', () => {
    const to = '/rota';
    const texto = 'Link de teste';
    const imagemSrc = 'icone-teste.png';

    render(
      <BrowserRouter>
        <MenuNavegacao to={to} texto={texto} imagemSrc={imagemSrc} />
      </BrowserRouter>
    );

    const linkElement = screen.getByText(texto);
    expect(linkElement).toBeInTheDocument();

    const imagemElement = screen.getByAltText(`icone de ${texto}`);
    expect(imagemElement).toBeInTheDocument();
    expect(imagemElement).toHaveAttribute(
      'src',
      `../imagens/icones/${imagemSrc}`
    );
  });

  test('Deve renderizar um link com a class link', () => {
    render(
      <BrowserRouter>
        <MenuNavegacao />
      </BrowserRouter>
    );

    const linkMenu = screen.getAllByRole('link');
    linkMenu.forEach((link) => {
    expect(link).toHaveClass('link');
    });
    expect(linkMenu).toMatchSnapshot();
  });
});
