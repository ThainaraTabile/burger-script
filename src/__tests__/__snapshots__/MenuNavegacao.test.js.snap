// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuNavegacao Deve renderizar um link com a class link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    <img
      alt="icone de undefined"
      class="icones"
      src="../imagens/icones/undefined"
    />
  </a>,
]
`;
